### PR TITLE
Investigate HA test failures

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1816

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)

## Discussion

-

## Testing

Running e2e tests.

## Documentation

-

## Follow-up

-

### CODE REVIEW

**Review by johscheuer — COMMENTED** [review]
I can see the two tests fail again, so we can use this run for debugging:

```text

Summarizing 2 Failures:
  [FAIL] Operator HA Upgrades upgrading a cluster with operator pod chaos and without foundationdb pod chaos [It] Upgrade from 7.1.37 to 7.3.15 [e2e, pr]
  /codebuild/output/src3248941900/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:197
  [FAIL] Operator HA Upgrades Upgrading a multi-DC cluster, with a temporary partition [It] Upgrade, with a temporary partition, from 7.1.37 to 7.3.15 [e2e, pr]
  /codebuild/output/src3248941900/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/fixtures/ha_fdb_cluster.go:313
```

File: e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go [hunk 1]
```diff
@@ -233,12 +233,12 @@ var _ = Describe("Operator HA Upgrades", Label("e2e", "pr"), func() {
 			// change, which happens a number of times, during an upgrade).
 			Expect(finalGeneration).To(BeNumerically("<=", initialGeneration+80))
 		},
-		EntryDescription("Upgrade from %[1]s to %[2]s"),
+		EntryDescription("upgrading from %[1]s to %[2]s"),
```
Comment (johscheuer) [inline e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:236]:
```suggestion
		EntryDescription("Upgrade from %[1]s to %[2]s"),
```

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [275bc54](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/275bc54fa24fd16fabdc333bbf75a31c506dee9d)
> * Duration 3:14:24
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/68566f3c-4625-48cd-a49c-1bc3549cdbca.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/68566f3c-4625-48cd-a49c-1bc3549cdbca/workspace.zip) zip file of the working directory (available for 30 days)

```text
[Container] 2023/09/29 13:58:47 Running command if $(grep -q -- "--- FAIL:" ${CODEBUILD_SRC_DIR}/logs/*.log); then echo "TESTS FAILED SEE THESE LOGS:"; echo ; grep -l -- "--- FAIL:" ${CODEBUILD_SRC_DIR}/logs/*.log; export fail_test=true; fi
TESTS FAILED SEE THESE LOGS:

/codebuild/output/src3225645483/src/github.com/FoundationDB/fdb-kubernetes-operator/logs/test_operator_upgrades.log
/codebuild/output/src3225645483/src/github.com/FoundationDB/fdb-kubernetes-operator/logs/test_operator_upgrades_variations.log
```

At least not HA upgrade failures 🎉

**Comment by johscheuer** [comment]
One failure:

```text
Operator Upgrades with locality based exclusions [It] Upgrade from 7.1.37 to 7.3.15 [e2e, pr]
/codebuild/output/src3225645483/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/fixtures/upgrade_test_configuration.go:105

  [FAILED] Unexpected error:
      <*errors.errorString | 0xc000329b10>: 
      timed out waiting for the condition
      {
          s: "timed out waiting for the condition",
      }
  occurred
  In [It] at: /codebuild/output/src3225645483/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_upgrades_variations/operator_upgrades_variations_test.go:361 @ 09/29/23 12:11:05.386

  There were additional failures detected.  To view them in detail run ginkgo -vv
```

and

```text
• [FAILED] [2494.356 seconds]
Operator Upgrades upgrading a cluster and one coordinator is not restarted [It] Upgrade from 7.1.37 to 7.3.15 with one coordinator not being restarted [e2e, pr]
/codebuild/output/src3225645483/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/fixtures/upgrade_test_configuration.go:105

  [FAILED] Unexpected error:
      <*errors.errorString | 0xc000321b40>: 
      timed out waiting for the condition
      {
          s: "timed out waiting for the condition",
      }
  occurred
  In [It] at: /codebuild/output/src3225645483/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_upgrades/operator_upgrades_test.go:318 @ 09/29/23 12:29:23.349

  There were additional failures detected.  To view them in detail run ginkgo -vv
```

looks like the cluster took longer to recover.

**Review by AnnigeriShambu — APPROVED** [review]

**Pull request merged**